### ISSUE TITLE
Add possibility to customize BaselinedScaledColumn with provided func

### ISSUE THREAD

**Issue by adamsitnik** [dotnet/BenchmarkDotNet#721]
Suggested by @stephentoub

>  I actually don't love Scaled, as it's not how I think about the results.  If something goes from 10ms to 7ms, scaled shows me 1.0 and 0.7, but I'd actually prefer a column that shows me 1.43x.  Is something like that built-in?
> I know there are various formulas folks like to use on different occasions, e.g. old/new, (old - new)/old, etc.  Might be interesting to explore parameterizing it with a func that would enable customizing the computation.
> You could maybe do that on the existing BaselinedScaledColumn... and then you wouldn't need a new name :)

**Comment by L-Dogg** [comment]
@adamsitnik Could I work on that issue?

**Comment by adamsitnik** [comment]
@L-Dogg sure!

**Comment by sasivishnu** [comment]
@adamsitnik Is this issue still open? can I take a look?

**Comment by adamsitnik** [comment]
@sasivishnu yes! please let me know if you need any help

**Comment by sasivishnu** [comment]
@adamsitnik Sure. I am out for this weekend. Will work on the issue from next week. Thanks.

**Comment by pedrofialho** [comment]
@adamsitnik is this issue still valid since `BaselineScaledColumn` is obsolete now in favor of `BaselineRatioColumn`?

**Comment by adamsitnik** [comment]
We have not made any progress on this since 2018 and I think that even @stephentoub got used to the `Ratio` column as I don't see him removing it from the results that he shares anymore ;)

Having said that, I am going to close the issue as it's unlikely that it would get implemented.

**Comment by stephentoub** [comment]
> I think that even @stephentoub got used to the Ratio column as I don't see him removing it from the results that he shares anymore ;)

Not really,  but I got tired of fighting it when I copy/paste it.  The current output is still almost never what I want.

**Comment by AndreyAkinshin** [comment]
@stephentoub do I understand correctly that you just need a column that displays "Inverted Ratio"?

**Comment by stephentoub** [comment]
> do I understand correctly that you just need a column that displays "Inverted Ratio"?

I don't "need" it, but yes, I'd like it :)  I want something that makes it easy to speak to an improvement.  If one row shown 100% and the other 15%, I find that difficult to speak to or think about how much of an improvement that was, whereas if those same rows showed 1.0x and 6.67x, respectively, I personally find that a lot easier to reason about, speak to in a demo, etc.

**Comment by AndreyAkinshin** [comment]
Ok, it makes sense. And what about benchmarks that are degraded? Should we always present "Inverted Ratio" or should we always present a number which is more than 1.0. We can consider the following summary table designs:

**Classic:**
```
| Method | Ratio |
|--------|-------|
| A      | 1.0   |
| B      | 0.15  |
| C      | 1.15  |
```

**Option A:**
```
| Method | InvRatio |
|--------|----------|
| A      | 1.0      |
| B      | 6.67     |
| C      | 0.87     |
```

Here we get a nice presentation for faster benchmarks, but the value for slower benchmarks look confusing.

**Option B:**
```
| Method | InvRatio     |
|--------|--------------|
| A      | baseline     |
| B      | 6.67x faster |
| C      | 1.15x slower |
```

Option B looks like the most meaningful one, it's very easy to intrepret it without the legend. However, the "faster/slower" postfix looks a little bit verbose, it makes the column much wider. Any ideas how to distinguish "slower" and "faster" benchmarks in a more compact way?

I was also thinking about the "percent" view for this column:

**Option C:**
```
| Method | Ratio    |
|--------|----------|
| A      | baseline |
| B      | -85%     |
| C      | +15%     |
```

If we evolve this notation, we can think about something like this:

**Option D:**
```
| Method | Ratio    |
|--------|----------|
| A      | baseline |
| B      | /6.67    |
| C      | *1.15    |
```

Option D looks like want you really want in a compact form, but I'm not sure that it would be obvious what's going on here.

**Comment by stephentoub** [comment]
Thanks, @AndreyAkinshin.

Option B is what I had in mind, and personally, I'm fine with the verbosity.

But if you want something shorter, how about using Unicode arrows (though that may not work in all terminals):
```
| Method | InvRatio     |
|--------|--------------|
| A      | baseline     |
| B      | 6.67x ↑      |
| C      | 1.15x ↓      |
```
or if you want to stick with ASCII, symbols that look like arrows:
```
| Method | InvRatio     |
|--------|--------------|
| A      | baseline     |
| B      | 6.67x ^      |
| C      | 1.15x v      |
```
or just shorter words, using "up" and "down":
```
| Method | InvRatio     |
|--------|--------------|
| A      | baseline     |
| B      | 6.67x up     |
| C      | 1.15x down   |
```
or +/-
```
| Method | InvRatio     |
|--------|--------------|
| A      | baseline     |
| B      | 6.67x +      |
| C      | 1.15x -      |
```

Personally I'm fine with any of these (as well as the original "faster"/"slower").

**Comment by AndreyAkinshin** [comment]
@stephentoub, implemented (this feature will be included in the upcoming BenchmarkDotNet v0.13.0).
Usage example:
```cs
[ShortRunJob, Config(typeof(Config))]
public class IntroRatioStyle
{
    [Benchmark(Baseline = true)]
    public void Baseline() => Thread.Sleep(1000);

    [Benchmark]
    public void Bar() => Thread.Sleep(150);

    [Benchmark]
    public void Foo() => Thread.Sleep(1150);

    private class Config : ManualConfig
    {
        public Config()
        {
            SummaryStyle = SummaryStyle.Default.WithRatioStyle(RatioStyle.Trend);
        }
    }
}
```
Output:
```markdown
|   Method |       Mean |   Error |  StdDev |        Ratio | RatioSD |
|--------- |-----------:|--------:|--------:|-------------:|--------:|
| Baseline | 1,000.6 ms | 2.48 ms | 0.14 ms |     baseline |         |
|      Bar |   150.9 ms | 1.30 ms | 0.07 ms | 6.63x faster |   0.00x |
|      Foo | 1,150.4 ms | 5.17 ms | 0.28 ms | 1.15x slower |   0.00x |
```
